android/libraries/rib-base/src/main/java/com/badoo/ribs/core/plugin/Plugin.kt

```diff
@@ -0,0 +1,69 @@
+package com.badoo.ribs.core.plugin
+
+import android.os.Bundle
+import android.view.ViewGroup
+import androidx.lifecycle.Lifecycle
+import com.badoo.ribs.core.Node
+import com.badoo.ribs.core.view.RibView
+
+interface Plugin
+
+interface NodeAware : Plugin {
+    fun init(node: Node<*>) {}
+}
+
+interface NodeLifecycleAware : Plugin {
+    fun onAttach(nodeLifecycle: Lifecycle) {}
+
+    fun onDetach() {}
+}
+
+interface ViewAware<V : RibView> : Plugin {
+    fun onViewCreated(view: V, viewLifecycle: Lifecycle) {}
+}
+
+interface ViewLifecycleAware : Plugin {
+    fun onAttachToView(parentViewGroup: ViewGroup) {}
+
+    fun onDetachFromView(parentViewGroup: ViewGroup) {}
+}
+
+interface SubtreeChangeAware : Plugin {
+    fun onAttachChildNode(child: Node<*>) {}
+
+    fun onDetachChildNode(child: Node<*>) {}
+}
+
+interface SubtreeViewChangeAware : Plugin {
+    fun onAttachChildView(child: Node<*>) {}
+
+    fun onDetachChildView(child: Node<*>) {}
+}
+
+interface AndroidLifecycleAware : Plugin {
```

Comment:
Both have further implications. Current interface doesn't mean change, just expressing already existing functionality via interface.

Otherwise good idea, but probably not in the scope of this PR.